src/SqlAsyncCollector.cs

```diff
@@ -423,23 +446,27 @@ public static async Task<TableInformation> RetrieveTableInformationAsync(SqlConn
                     await sqlConnection.CloseAsync();
                 }
 
-                if (!primaryKeys.Any())
+                if (!primaryKeys.Any(k => k.IsIdentity) && !primaryKeys.Any())
```

Comment:
Isn't `!primaryKeys.Any()` condition enough to check for primary keys here? Do we need the `!primary.Any(k => k.IsIdentity)` check as well?